moveit_setup_assistant/src/widgets/group_edit_widget.cpp

```diff
@@ -249,6 +264,33 @@ void GroupEditWidget::setSelected(const std::string& group_name)
   }
   kinematics_timeout_field_->setText(QString::number(*timeout));
 
+  // Load goal joint tolerance
+  double* goal_joint_tolerance = &config_data_->group_meta_data_[group_name].goal_joint_tolerance_;
+  if (*goal_joint_tolerance == 0)
+  {
+    // Set default value
+    *goal_joint_tolerance = DEFAULT_GOAL_JOINT_TOLERANCE;
+  }
+  goal_joint_tolerance_field_->setText(QString::number(*goal_joint_tolerance));
```

Comment:
I suggest replacing the pointer with a reference:
```diff
-double* goal_joint_tolerance = &config_data_->group_meta_data_[group_name].goal_joint_tolerance_;
+double& goal_joint_tolerance = config_data_->group_meta_data_[group_name].goal_joint_tolerance_; 
```

In general, try to avoid code redundancy, e.g. by introducing a lambda function:
```cpp
auto setValue = [](double& value, const double dflt, QDoubleSpinBox *widget) {
  if (value <= 0)
    value = dflt;
  widget->setValue(value);
};
setValue(config_data_->group_meta_data_[group_name].goal_joint_tolerance_, DEFAULT_GOAL_JOINT_TOLERANCE, goal_joint_tolerance_field_);
```